examples/autoignition-mpi.py

```diff
@@ -358,6 +366,23 @@ def my_health_check(dv):
 
         return health_error
 
+    def my_get_timestep(t, dt, state):
+        #  richer interface to calculate {dt,dfl} returns node-local estimates
```

Comment:
```suggestion
        #  richer interface to calculate {dt,cfl} returns node-local estimates
```

Reply:
Out of curiosity: Do you think it would be worth adding short docstrings to these `my_foo(...)` functions used in the callback routines? Nothing super formal, just a short comment explaining what it does.

Reply:
I think that code comments would work fine for those;  I am not sure about adding docstrings for the driver things that aren't in the library. That said, I am not opposed at all if you think they should be there.  Since they are completely user option - I feel a little funny about legitimizing them by adding docs.

What I do really think we need, is documentation that explains the callbacks, including the cautionary statements about state modification - and possibly including examples of callbacks from the example drivers.

Reply:
FYI, I applied your suggestion but unresolved cause we were talking here.  How rude, github.

Reply:
> What I do really think we need, is documentation that explains the callbacks, including the cautionary statements about state modification - and possibly including examples of callbacks from the example drivers.

Fair enough. That can be done in a separate PR. I'm totally fine with that as an alternative.

Reply:
#427